lib/engine.js

```diff
@@ -18,132 +18,139 @@
 'use strict';
 
 var path = require('path'),
-    dust = require('dustjs-linkedin'),
     utils = require('./utils'),
-    patch = require('./patch'),
-    River = require('./brook'),
+    Brook = require('./brook'),
     reqwire = require('./reqwire');
+var VError = require('verror');
 
+var isAbsolute = require('path-is-absolute');
 
-var loaders = {
-    dust: require('./reader/dust'),
-    js: require('./reader/js')
-};
-
+var fs = require('fs');
 
+var debug = require('debuglog')('adaro');
 
-function createRenderContext(options) {
-    var context;
+var freshy = require('freshy').freshy;
 
-    // Hoist *only* view, settings, and ext to global context.
-    context = dust.makeBase({
-        views: options.views,
-        settings: options.settings,
-        ext: options.ext
-    });
 
-    // Remove from child context so global values are used.
-    delete options.views;
-    delete options.settings;
-    delete options.ext;
+exports.create = function (type, config) {
+    var engine;
 
-    context = context.push(options);
-    return context;
-}
+    config = config || {};
+    config.cache = (config.cache === undefined) ? true : !!config.cache;
 
 
+    var dust = freshy('dustjs-linkedin', function (dust) {
+        for (var k in config) {
+            dust.config[k] = config[k];
+        }
 
-exports.create = function (type, config) {
-    var reader, engine;
+        // Load referenced helpers
+        if (Array.isArray(config.helpers)) {
+            config.helpers.forEach(function (module) {
+                // allow configuration of helpers like:
+                // "helpers": ["./lib/abc", { "name": "./lib/test", "arguments": { "debug": true } }] 
+                if (typeof module === 'function') {
+                    return module(dust);
+                }
+                if (typeof module === 'string') {
+                    reqwire.init(module, dust);
+                }
+                else if (module.hasOwnProperty('name')) {
+                    reqwire.init(module.name, dust, module.arguments);
+                }
+            });
+        }
 
-    config = config || {};
-    config.cache = (config.cache === undefined) ? true : !!config.cache;
+    });
 
-    // Load referenced helpers
-    if (Array.isArray(config.helpers)) {
-        config.helpers.forEach(function (module) {
-            // allow configuration of helpers like:
-            // "helpers": ["./lib/abc", { "name": "./lib/test", "arguments": { "debug": true } }] 
-            if (typeof module === 'string') {
-                reqwire.init(module, dust);
+    dust.onLoad = function (name, options, cb) {
+        if (isAbsolute(name[0])) {
+            debug("loading '%s'", name);
+            read(null, name);
+        } else if (options.view && options.view.lookup && options.view.lookup.length >= 3) {
```

Comment:
Good question! I need to go think through it again! Whether or not the extension is included is super inconsistent in some of these APIs.